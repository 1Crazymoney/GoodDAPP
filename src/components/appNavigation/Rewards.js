import React from 'react'
import { View, Text } from 'react-native'
import { Link } from '@react-navigation/web'
import { createStackNavigator, PushButton } from './stackNavigation'
class InternalReward2 extends React.Component {
  // This is one way of setting params for screen
  // More useful if the params are the same for all instances of this component regarding how is included in the stack
  static navigationOptions = {
    title: 'Internal Reward Screen 2'
  }

  render() {
    return (
      <View>
        <Text>Internal Reward Screen 2 </Text>
      </View>
    )
  }
}

const InternalReward1 = props => (
  <View>
    <Text>InternalReward Screen 1</Text>
    <PushButton routeName={'InternalReward2'} navigationConfig={props.screenProps}>
      InternalReward 2
    </PushButton>
  </View>
)

const Rewards = props => {
  console.log('Rewards props', props)
  return (
    <View>
      <Text>Rewards Screen</Text>
      <PushButton routeName={'InternalReward1'} navigationConfig={props.screenProps}>
        InternalReward 1
      </PushButton>
    </View>
  )
}

export default createStackNavigator({
  Rewards,
  InternalReward1: {
    screen: InternalReward1,
    // This is one way of setting params for screen.
    // More useful if the params may change for the same component
    navigationOptions: { title: 'Internal Reward Title 1' }
  },
  InternalReward2
})
