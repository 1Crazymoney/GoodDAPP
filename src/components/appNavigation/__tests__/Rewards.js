import React from 'react'
import Rewards from '../Rewards'
import { createSwitchNavigator } from '@react-navigation/core'
import { createBrowserApp } from '@react-navigation/web'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

describe('Rewards', () => {
  it('renders without errors', () => {
    const WebRouter = createBrowserApp(createSwitchNavigator({ Rewards }))

    const tree = renderer.create(<WebRouter />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const WebRouter = createBrowserApp(createSwitchNavigator({ Rewards }))

    const component = renderer.create(<WebRouter />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
