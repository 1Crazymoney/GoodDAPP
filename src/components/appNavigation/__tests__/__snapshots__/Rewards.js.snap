// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rewards matches snapshot 1`] = `
Array [
  <div
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitBoxAlign": "center",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "horizontal",
        "WebkitBoxPack": "center",
        "WebkitFlexBasis": "auto",
        "WebkitFlexDirection": "row",
        "WebkitFlexShrink": 0,
        "WebkitJustifyContent": "center",
        "alignItems": "center",
        "backgroundColor": "rgba(144,144,144,1.00)",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0px",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0px",
        "borderRightStyle": "solid",
        "borderRightWidth": "0px",
        "borderTopStyle": "solid",
        "borderTopWidth": "0px",
        "boxSizing": "border-box",
        "display": "flex",
        "flexBasis": "auto",
        "flexDirection": "row",
        "flexShrink": 0,
        "height": "60px",
        "justifyContent": "center",
        "marginBottom": "0px",
        "marginLeft": "0px",
        "marginRight": "0px",
        "marginTop": "0px",
        "minHeight": "0px",
        "minWidth": "0px",
        "msFlexAlign": "center",
        "msFlexDirection": "row",
        "msFlexNegative": 0,
        "msFlexPack": "center",
        "msFlexPreferredSize": "auto",
        "paddingBottom": "0px",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "paddingTop": "0px",
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      style={
        Object {
          "WebkitAlignItems": "stretch",
          "WebkitBoxAlign": "stretch",
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "vertical",
          "WebkitFlexBasis": "auto",
          "WebkitFlexDirection": "column",
          "WebkitFlexShrink": 0,
          "alignItems": "stretch",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0px",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0px",
          "borderRightStyle": "solid",
          "borderRightWidth": "0px",
          "borderTopStyle": "solid",
          "borderTopWidth": "0px",
          "boxSizing": "border-box",
          "display": "flex",
          "flexBasis": "auto",
          "flexDirection": "column",
          "flexShrink": 0,
          "marginBottom": "0px",
          "marginLeft": "0px",
          "marginRight": "0px",
          "marginTop": "0px",
          "minHeight": "0px",
          "minWidth": "40px",
          "msFlexAlign": "stretch",
          "msFlexDirection": "column",
          "msFlexNegative": 0,
          "msFlexPreferredSize": "auto",
          "paddingBottom": "10px",
          "paddingLeft": "10px",
          "paddingRight": "10px",
          "paddingTop": "10px",
          "position": "relative",
          "zIndex": 0,
        }
      }
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "WebkitAlignItems": "stretch",
            "WebkitBoxAlign": "stretch",
            "WebkitBoxDirection": "normal",
            "WebkitBoxOrient": "vertical",
            "WebkitFlexBasis": "auto",
            "WebkitFlexDirection": "column",
            "WebkitFlexShrink": 0,
            "alignItems": "stretch",
            "backgroundImage": "url(backButton.png)",
            "borderBottomStyle": "solid",
            "borderBottomWidth": "0px",
            "borderLeftStyle": "solid",
            "borderLeftWidth": "0px",
            "borderRightStyle": "solid",
            "borderRightWidth": "0px",
            "borderTopStyle": "solid",
            "borderTopWidth": "0px",
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "flex",
            "flexBasis": "auto",
            "flexDirection": "column",
            "flexShrink": 0,
            "height": "25px",
            "marginBottom": "0px",
            "marginLeft": "0px",
            "marginRight": "0px",
            "marginTop": "0px",
            "minHeight": "0px",
            "minWidth": "0px",
            "msFlexAlign": "stretch",
            "msFlexDirection": "column",
            "msFlexNegative": 0,
            "msFlexPreferredSize": "auto",
            "paddingBottom": "0px",
            "paddingLeft": "0px",
            "paddingRight": "0px",
            "paddingTop": "0px",
            "position": "relative",
            "width": "25px",
            "zIndex": 0,
          }
        }
      />
    </div>
    <div
      dir="auto"
      style={
        Object {
          "WebkitBoxFlex": 2,
          "WebkitFlexGrow": 2,
          "borderBottomWidth": "0px",
          "borderLeftWidth": "0px",
          "borderRightWidth": "0px",
          "borderTopWidth": "0px",
          "boxSizing": "border-box",
          "color": "rgba(210,210,210,1.00)",
          "display": "inline",
          "flexGrow": 2,
          "fontFamily": "system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Ubuntu, \\"Helvetica Neue\\", sans-serif",
          "fontSize": "18px",
          "fontStyle": "inherit",
          "fontVariant": "inherit",
          "fontWeight": "inherit",
          "lineHeight": "inherit",
          "marginBottom": "0px",
          "marginLeft": "0px",
          "marginRight": "0px",
          "marginTop": "0px",
          "msFlexPositive": 2,
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
          "textAlign": "center",
          "textDecoration": "none",
          "textTransform": "uppercase",
          "whiteSpace": "pre-wrap",
          "wordWrap": "break-word",
        }
      }
    >
      Rewards
    </div>
    <div
      style={
        Object {
          "WebkitAlignItems": "stretch",
          "WebkitBoxAlign": "stretch",
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "vertical",
          "WebkitFlexBasis": "auto",
          "WebkitFlexDirection": "column",
          "WebkitFlexShrink": 0,
          "alignItems": "stretch",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0px",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0px",
          "borderRightStyle": "solid",
          "borderRightWidth": "0px",
          "borderTopStyle": "solid",
          "borderTopWidth": "0px",
          "boxSizing": "border-box",
          "display": "flex",
          "flexBasis": "auto",
          "flexDirection": "column",
          "flexShrink": 0,
          "marginBottom": "0px",
          "marginLeft": "0px",
          "marginRight": "0px",
          "marginTop": "0px",
          "minHeight": "0px",
          "minWidth": "40px",
          "msFlexAlign": "stretch",
          "msFlexDirection": "column",
          "msFlexNegative": 0,
          "msFlexPreferredSize": "auto",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
          "position": "relative",
          "zIndex": 0,
        }
      }
    />
  </div>,
  <div
    style={
      Object {
        "WebkitAlignItems": "stretch",
        "WebkitBoxAlign": "stretch",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "vertical",
        "WebkitFlexBasis": "auto",
        "WebkitFlexDirection": "column",
        "WebkitFlexShrink": 0,
        "alignItems": "stretch",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0px",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0px",
        "borderRightStyle": "solid",
        "borderRightWidth": "0px",
        "borderTopStyle": "solid",
        "borderTopWidth": "0px",
        "boxSizing": "border-box",
        "display": "flex",
        "flexBasis": "auto",
        "flexDirection": "column",
        "flexShrink": 0,
        "marginBottom": "0px",
        "marginLeft": "0px",
        "marginRight": "0px",
        "marginTop": "0px",
        "minHeight": "0px",
        "minWidth": "0px",
        "msFlexAlign": "stretch",
        "msFlexDirection": "column",
        "msFlexNegative": 0,
        "msFlexPreferredSize": "auto",
        "paddingBottom": "0px",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "paddingTop": "0px",
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      dir="auto"
      style={
        Object {
          "borderBottomWidth": "0px",
          "borderLeftWidth": "0px",
          "borderRightWidth": "0px",
          "borderTopWidth": "0px",
          "boxSizing": "border-box",
          "color": "inherit",
          "display": "inline",
          "fontFamily": "system-ui, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Ubuntu, \\"Helvetica Neue\\", sans-serif",
          "fontSize": "14px",
          "fontStyle": "inherit",
          "fontVariant": "inherit",
          "fontWeight": "inherit",
          "lineHeight": "inherit",
          "marginBottom": "0px",
          "marginLeft": "0px",
          "marginRight": "0px",
          "marginTop": "0px",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
          "textDecoration": "none",
          "whiteSpace": "pre-wrap",
          "wordWrap": "break-word",
        }
      }
    >
      Rewards Screen
    </div>
    <div
      style={
        Object {
          "WebkitAlignItems": "stretch",
          "WebkitBoxAlign": "stretch",
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "horizontal",
          "WebkitBoxPack": "justify",
          "WebkitFlexBasis": "auto",
          "WebkitFlexDirection": "row",
          "WebkitFlexShrink": 0,
          "WebkitJustifyContent": "space-around",
          "alignItems": "stretch",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0px",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0px",
          "borderRightStyle": "solid",
          "borderRightWidth": "0px",
          "borderTopStyle": "solid",
          "borderTopWidth": "0px",
          "boxSizing": "border-box",
          "display": "flex",
          "flexBasis": "auto",
          "flexDirection": "row",
          "flexShrink": 0,
          "justifyContent": "space-around",
          "marginBottom": "0px",
          "marginLeft": "0px",
          "marginRight": "0px",
          "marginTop": "0px",
          "minHeight": "0px",
          "minWidth": "0px",
          "msFlexAlign": "stretch",
          "msFlexDirection": "row",
          "msFlexNegative": 0,
          "msFlexPack": "distribute",
          "msFlexPreferredSize": "auto",
          "paddingBottom": "10px",
          "paddingLeft": "10px",
          "paddingRight": "10px",
          "paddingTop": "10px",
          "position": "relative",
          "zIndex": 0,
        }
      }
    >
      <div
        style={
          Object {
            "WebkitAlignItems": "stretch",
            "WebkitBoxAlign": "stretch",
            "WebkitBoxDirection": "normal",
            "WebkitBoxOrient": "vertical",
            "WebkitFlexBasis": "auto",
            "WebkitFlexDirection": "column",
            "WebkitFlexShrink": 0,
            "alignItems": "stretch",
            "backgroundColor": "rgba(85,85,85,1.00)",
            "borderBottomColor": "rgba(0,0,0,0.00)",
            "borderBottomLeftRadius": "4px",
            "borderBottomRightRadius": "4px",
            "borderBottomStyle": "solid",
            "borderBottomWidth": "0px",
            "borderLeftColor": "rgba(0,0,0,0.00)",
            "borderLeftStyle": "solid",
            "borderLeftWidth": "0px",
            "borderRightColor": "rgba(0,0,0,0.00)",
            "borderRightStyle": "solid",
            "borderRightWidth": "0px",
            "borderTopColor": "rgba(0,0,0,0.00)",
            "borderTopLeftRadius": "4px",
            "borderTopRightRadius": "4px",
            "borderTopStyle": "solid",
            "borderTopWidth": "0px",
            "boxShadow": "0px NaNpx 2px rgba(0,0,0,0.24)",
            "boxSizing": "border-box",
            "display": "flex",
            "flexBasis": "auto",
            "flexDirection": "column",
            "flexShrink": 0,
            "marginBottom": "10px",
            "marginLeft": "0px",
            "marginRight": "0px",
            "marginTop": "0px",
            "minHeight": "0px",
            "minWidth": "64px",
            "msFlexAlign": "stretch",
            "msFlexDirection": "column",
            "msFlexNegative": 0,
            "msFlexPreferredSize": "auto",
            "paddingBottom": "5px",
            "paddingLeft": "0px",
            "paddingRight": "0px",
            "paddingTop": "5px",
            "position": "relative",
            "zIndex": 0,
          }
        }
      >
        <div
          data-focusable={true}
          disabled={false}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          role="button"
          style={
            Object {
              "MozUserSelect": "none",
              "WebkitAlignItems": "stretch",
              "WebkitBoxAlign": "stretch",
              "WebkitBoxDirection": "normal",
              "WebkitBoxOrient": "vertical",
              "WebkitFlexBasis": "auto",
              "WebkitFlexDirection": "column",
              "WebkitFlexShrink": 0,
              "WebkitUserSelect": "none",
              "alignItems": "stretch",
              "backgroundColor": "rgba(0,0,0,0.00)",
              "borderBottomLeftRadius": "4px",
              "borderBottomRightRadius": "4px",
              "borderBottomStyle": "solid",
              "borderBottomWidth": "0px",
              "borderLeftStyle": "solid",
              "borderLeftWidth": "0px",
              "borderRightStyle": "solid",
              "borderRightWidth": "0px",
              "borderTopLeftRadius": "4px",
              "borderTopRightRadius": "4px",
              "borderTopStyle": "solid",
              "borderTopWidth": "0px",
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "flex",
              "flexBasis": "auto",
              "flexDirection": "column",
              "flexShrink": 0,
              "marginBottom": "0px",
              "marginLeft": "0px",
              "marginRight": "0px",
              "marginTop": "0px",
              "minHeight": "0px",
              "minWidth": "0px",
              "msFlexAlign": "stretch",
              "msFlexDirection": "column",
              "msFlexNegative": 0,
              "msFlexPreferredSize": "auto",
              "msTouchAction": "manipulation",
              "msUserSelect": "none",
              "paddingBottom": "0px",
              "paddingLeft": "0px",
              "paddingRight": "0px",
              "paddingTop": "0px",
              "position": "relative",
              "touchAction": "manipulation",
              "userSelect": "none",
              "zIndex": 0,
            }
          }
          tabIndex="0"
        >
          <div
            style={
              Object {
                "WebkitAlignItems": "center",
                "WebkitBoxAlign": "center",
                "WebkitBoxDirection": "normal",
                "WebkitBoxOrient": "horizontal",
                "WebkitBoxPack": "center",
                "WebkitFlexBasis": "auto",
                "WebkitFlexDirection": "row",
                "WebkitFlexShrink": 0,
                "WebkitJustifyContent": "center",
                "alignItems": "center",
                "borderBottomStyle": "solid",
                "borderBottomWidth": "0px",
                "borderLeftStyle": "solid",
                "borderLeftWidth": "0px",
                "borderRightStyle": "solid",
                "borderRightWidth": "0px",
                "borderTopStyle": "solid",
                "borderTopWidth": "0px",
                "boxSizing": "border-box",
                "display": "flex",
                "flexBasis": "auto",
                "flexDirection": "row",
                "flexShrink": 0,
                "justifyContent": "center",
                "marginBottom": "0px",
                "marginLeft": "0px",
                "marginRight": "0px",
                "marginTop": "0px",
                "minHeight": "0px",
                "minWidth": "0px",
                "msFlexAlign": "center",
                "msFlexDirection": "row",
                "msFlexNegative": 0,
                "msFlexPack": "center",
                "msFlexPreferredSize": "auto",
                "paddingBottom": "0px",
                "paddingLeft": "0px",
                "paddingRight": "0px",
                "paddingTop": "0px",
                "position": "relative",
                "zIndex": 0,
              }
            }
          >
            <div
              dir="auto"
              style={
                Object {
                  "borderBottomWidth": "0px",
                  "borderLeftWidth": "0px",
                  "borderRightWidth": "0px",
                  "borderTopWidth": "0px",
                  "boxSizing": "border-box",
                  "color": "rgba(255,255,255,1.00)",
                  "direction": "ltr",
                  "display": "inline",
                  "fontFamily": "Roboto-Medium",
                  "fontSize": "14px",
                  "fontStyle": "inherit",
                  "fontVariant": "inherit",
                  "fontWeight": "inherit",
                  "letterSpacing": "1px",
                  "lineHeight": "inherit",
                  "marginBottom": "9px",
                  "marginLeft": "16px",
                  "marginRight": "16px",
                  "marginTop": "9px",
                  "maxWidth": "100%",
                  "overflowX": "hidden",
                  "overflowY": "hidden",
                  "paddingBottom": "0px",
                  "paddingLeft": "0px",
                  "paddingRight": "0px",
                  "paddingTop": "0px",
                  "textAlign": "center",
                  "textDecoration": "none",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                  "wordWrap": "break-word",
                }
              }
            >
              <span
                dir="auto"
                style={
                  Object {
                    "borderBottomWidth": "0px",
                    "borderLeftWidth": "0px",
                    "borderRightWidth": "0px",
                    "borderTopWidth": "0px",
                    "boxSizing": "border-box",
                    "color": "rgba(255,255,255,1.00)",
                    "direction": "ltr",
                    "display": "inline",
                    "fontFamily": "Helvetica, \\"sans-serif\\"",
                    "fontSize": "18px",
                    "fontStyle": "inherit",
                    "fontVariant": "inherit",
                    "fontWeight": "bold",
                    "lineHeight": "inherit",
                    "marginBottom": "0px",
                    "marginLeft": "0px",
                    "marginRight": "0px",
                    "marginTop": "0px",
                    "paddingBottom": "10px",
                    "paddingLeft": "10px",
                    "paddingRight": "10px",
                    "paddingTop": "10px",
                    "textAlign": "center",
                    "textDecoration": "none",
                    "textTransform": "uppercase",
                    "whiteSpace": "inherit",
                    "wordWrap": "break-word",
                  }
                }
              >
                InternalReward 1
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
